### DISCUSSION TITLE
Stream audio on wireless headset throught BLE?

### DISCUSSION THREAD

**Discussion by feldmarL** [pschatzmann/arduino-audio-tools#678] [Q&A] [answered]
Hey,

A very simple question which may sounds stupid but I can't found a simple yes / no answer to it.

Is it possible, using only a native Arduino Nano 33 BLE, to stream low quality and short audio files on a wireless BLE compatible headset?

Thanks in advance!

**Comment by pschatzmann** [comment]
I don't know! To my knowledge these headsets use HFP or HSP, but this has nothing to do with BLE!
So the answer would be no: These protocols are part of classical bluetooth which is not supported by the Nano!

**Reply by feldmarL** [reply]
So is there to your knowledge a Bluetooth speaker (of any kind, really!) that could play an audio file sent from an arduino of this type?

Trying hard to do this for a school project, a POC using any type of speaker would be great!

**Comment by pschatzmann** [comment, marked as answer]
I would suggest to replace the Arduino Nano 33 BLE with an ESP32. This processor is cheaper and then you could use my [A2DP library](https://github.com/pschatzmann/ESP32-A2DP).

If you need to stick with a RP2040, then I would suggest to find with what mechanism you get a max transmission speed and then build a sender and receiver. E.g. you could try to use [Wifi](https://www.pschatzmann.ch/home/2023/02/28/arduino-audiotools-esp32-vs-rp2040-pico-w/) 

But in any case, this would not involve any Headsets!

**Reply by feldmarL** [reply]
You mention I may want to "stick with a RP2040" but didn't mentionned it earlier. Also it seems it could be a viable alternative?

You also say that it would not involve any headset. Your A2DP library can't be used to read an audio file (a simple file on an sd card connected to the board in my case) to a remote headset (or any kind of bluetooth speaker)?

I may be totally wrong here or not understanding a key point?

Already a big thanks for the time you gave to my questions!

**Reply by pschatzmann** [reply]
My mistake: I mixed it up with the Arduino Nano 33 BLE! For both processors the implementation from Arduino is based on MBED.

Headset no - blutooth speaker and earphones yes. 
What exactly do you mean when you talk about headset ?

**Reply by feldmarL** [reply]
To add a little more context:

I'm working on an IoT white cane (for blind users) from which I want to stream audio file through bluetooth. We've been using Arduino Nano 33 BLE so far for all of the other characteristics such as fall detection, geolocation etc.

We chosed this board because we initially thought that BLE was as simple as a technology compatible with classic Bluetooth using less energy. Now facing a wall when trying to do what we want to do.

English is not my main language, I may have used headset wrongly so here is the explanation of what I want:
I meant a wearable bluetooth device used as a speaker such as in my case Beats Studio 3 or any earphone really. Hope this clears the path.

We think we could easily adapt the other functionnality with an other board but we simple now want to make a great, if not the best, decision for the continuation. You've been of precious and fast help, immense thanks for that!

**Reply by pschatzmann** [reply]
I think you can pretty much forget about audio output using the Arduino Nano 33 BLE...
If I remember right, I never managed to have I2S working and the best I managed was some simulated audio using PWM

**Reply by feldmarL** [reply]
Great, that's a way to start then.

Is there a board you would recommend for this task? We may also just get rid of it and use another system but we really want to try it out. Lmk!

**Reply by pschatzmann** [reply]
The best is an ESP32 with my [A2DP library](https://github.com/pschatzmann/ESP32-A2DP).
But be warned, this is one of the more challenging scenarios...

**Reply by feldmarL** [reply]
Then we'll try it, you may heard from us in a near future.

Thanks again for your time and consideration!

**Comment by pschatzmann** [comment]
You might also consider this: https://www.hackster.io/news/pi-pico-sdk-1-5-0-release-adds-bluetooth-support-7fe4d09da432

**Comment by pschatzmann** [comment]
And I found this: https://arduino-pico.readthedocs.io/en/latest/bluetooth.html
